pkg/types/gcp/machinepools.go

```diff
@@ -22,6 +24,15 @@ type MachinePool struct {
 	//
 	// +optional
 	Tags []string `json:"tags,omitempty"`
+
+	// OnHostMaintenance determines the behavior when a maintenance event occurs that might cause the instance to reboot.
+	// +optional
+	OnHostMaintenance v1beta1.GCPHostMaintenanceType `json:"onHostMaintenance,omitempty"`
```

Comment:
secure boot? disk encryption?

Reply:
disk encription is already implemented under `GCPDisk`
secure boot is something we will need to add to the GCPMachineProviderSpec

Reply:
@patrickdillon do we even support secure boot on gcp?
@fabiendupont